### DISCUSSION TITLE
I'm unclear on how to get iml autocomplete to work

### DISCUSSION THREAD

**Discussion by chrisdrackett** [dotansimha/graphql-code-generator#5917] [Q&A] [answered]
How do I teach VSCode to autocomplete with my config.yml?

**Comment by ardatan** [comment, marked as answer]
If you want to have autocomplete while editing configuration file, you can try this extension;
https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml

**Reply by chrisdrackett** [reply]
I have this extension, but I don't understand how to teach it that a given yml file should get graphql-codegen completion. At the moment it isn't working.

**Reply by ardatan** [reply]
The configuration file name should matche one of these;
https://github.com/SchemaStore/schemastore/blob/master/src/api/json/catalog.json#L1119
cc @dotansimha

**Reply by chrisdrackett** [reply]
yeah, this matches, but it isn't in the root of my projects (this is a monorepo) maybe that is the issue?

**Reply by ardatan** [reply]
I'm not sure how that extension actually works. Have you ever checked if it works with other yaml config files? Maybe it is something on your local.

**Reply by chrisdrackett** [reply]
I'm unclear how vscode/the extension knows what it is looking for. How does graphql-code-generator tell it what it handles?

**Reply by ardatan** [reply]
GraphQL Code Generator doesn't actually interact with with that extension in any ways.
It just takes our config's schema from this URL;
https://github.com/SchemaStore/schemastore/blob/edaf36090c4faf2deb91107b438824d0b2942999/src/api/json/catalog.json#L1118
Then does a validation based on that JSON Schema.
